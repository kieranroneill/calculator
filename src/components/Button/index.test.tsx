import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';

// Components.
import { Button, InputKeyEnum, Props } from './';

interface Scope {
    props: Props;
    wrapper: ShallowWrapper;
}

describe('src/components/Button', () => {
    let scope: Scope;

    beforeEach(() => {
        const props: Props = {
            label: InputKeyEnum.Addition,
            onClick: jest.fn(),
            purpose: 'numeric',
        };

        scope = {
            props,
            wrapper: shallow(<Button {...props} />),
        };
    });

    describe('<Button /> snapshots', () => {
        it('should match the snapshot with the default props', () => {
            expect(scope.wrapper).toMatchSnapshot();
        });

        it('should match the snapshot when the button is an operator', () => {
            scope.wrapper.setProps({
                purpose: 'operator',
            } as Props);

            expect(scope.wrapper).toMatchSnapshot();
        });

        it('should match the snapshot when the button is a result', () => {
            scope.wrapper.setProps({
                purpose: 'result',
            } as Props);

            expect(scope.wrapper).toMatchSnapshot();
        });
    });

    describe('when clicking the button', () => {
        it('should return the input', () => {
            scope.wrapper.simulate('click');

            expect(scope.props.onClick).toBeCalledWith(scope.props.label);
        });
    });
});
