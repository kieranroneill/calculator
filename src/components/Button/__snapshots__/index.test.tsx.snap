// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Button <Button /> snapshots should match the snapshot when the button is an executtion 1`] = `
<styled.button
  purpose="execute"
>
  +
</styled.button>
`;

exports[`src/components/Button <Button /> snapshots should match the snapshot when the button is an operator 1`] = `
<styled.button
  purpose="operator"
>
  +
</styled.button>
`;

exports[`src/components/Button <Button /> snapshots should match the snapshot with the default props 1`] = `
<styled.button
  purpose="numeric"
>
  +
</styled.button>
`;
